pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -550,10 +581,21 @@ func (r *ReconcileIntrusionDetection) setDefaultsOnIntrusionDetection(ctx contex
 			},
 		}
 
-		if err := r.client.Update(ctx, ids); err != nil {
-			return err
+	}
+
+	if ids.Spec.AnomalyDetectionSpec == (operatorv1.AnomalyDetectionSpec{}) {
```

Comment:
I think this can be rewritten to:
```
	if len(ids.Spec.AnomalyDetectionSpec.StorageType) == 0 {
		ids.Spec.AnomalyDetectionSpec.StorageType = operatorv1.EphemeralStorageType
	}
```